backend/dataall/modules/shares_base/db/share_object_repositories.py

```diff
@@ -195,56 +171,330 @@ def get_all_share_items_in_share(session, share_uri, status=None, healthStatus=N
+                ShareObject.principalId.label('principalId'),
+                ShareObject.principalType.label('principalType'),
+                ShareObject.environmentUri.label('targetEnvironmentUri'),
+                ShareObjectItem.itemType.label('itemType'),
+                ShareObjectItem.itemName.label('itemName'),
+                Organization.organizationUri.label('organizationUri'),
+                Organization.name.label('organizationName'),
+            )
+            .join(
+                ShareObject,
+                ShareObject.shareUri == ShareObjectItem.shareUri,
+            )
+            .join(
+                DatasetBase,
+                ShareObject.datasetUri == DatasetBase.datasetUri,
+            )
+            .join(
+                Environment,
+                Environment.environmentUri == DatasetBase.environmentUri,
+            )
+            .join(
+                Organization,
+                Organization.organizationUri == Environment.organizationUri,
+            )
             .filter(
                 and_(
-                    ShareObjectItem.shareUri == share.shareUri,
-                    ShareObjectItem.status.in_([ShareItemStatus.PendingApproval.value]),
+                    ShareObjectItem.status.in_(share_item_shared_states),
+                    ShareObject.environmentUri == env_uri,
                 )
             )
-            .all()
         )
-        if shared_items:
-            return True
-        return False
+
+        if data.get('datasetUri'):
+            dataset_uri = data.get('datasetUri')
+            q = q.filter(ShareObject.datasetUri == dataset_uri)
+
+        if data.get('itemTypes', None):
+            item_types = data.get('itemTypes')
+            q = q.filter(or_(*[ShareObjectItem.itemType == t for t in item_types]))
+
+        if data.get('uniqueShares', False):
+            q = q.filter(ShareObject.principalType != PrincipalType.ConsumptionRole.value)
+            q = q.order_by(ShareObject.shareUri).distinct(ShareObject.shareUri)
+        else:
+            q = q.order_by(ShareObjectItem.itemName).distinct()
+
+        if data.get('term'):
+            term = data.get('term')
+            q = q.filter(ShareObjectItem.itemName.ilike('%' + term + '%'))
+
+        return paginate(query=q, page=data.get('page', 1), page_size=data.get('pageSize', 10)).to_dict()
+
+    @staticmethod
+    def list_shareable_items_of_type(session, share, type, share_type_model, share_type_uri, status=None):
+        logger.info(f'Getting all shareable items {status=}, for {share_type_model=}')
+        query = (
+            session.query(
+                share_type_uri.label('itemUri'),
+                share_type_model.datasetUri.label('datasetUri'),
+                func.coalesce(type.value).label('itemType'),
+                share_type_model.description.label('description'),
+                share_type_model.name.label('itemName'),
```

Comment:
share_type_model.name.label('itemName') ---> this field replaces DatasetBucket.S3BucketName.label('itemName').... that's the reason behind the migration script